### PR TITLE
feat(argo-workflows): add optional HPA to Server

### PR DESCRIPTION
**Pull request by agilgur5**
## Summary

Best practice to have an HPA for the Argo Server for HA. Docs on the topic: https://argoproj.github.io/argo-workflows/high-availability/#argo-server

## Details

- Similar to the existing HPA for Argo CD, with slight modifications for AWF naming etc
  - Based off #240 except updated to latest version of templates

## Follow-ups

1. Add an optional VPA for the AWF Controller? (as it [can't scale horizontally](https://argoproj.github.io/argo-workflows/scaling/#horizontally-scaling))
    * I can add that myself as well if wanted, but I know the Controller requires a bit more precision to scale, e.g. running a [hot-standby](https://argoproj.github.io/argo-workflows/high-availability/#workflow-controller) (`replicas` set to `2`) and swapping between the two while the Pod goes down to scale vertically
      * [Example VPA Helm template from Prometheus](https://github.com/prometheus-community/helm-charts/blob/240daaf9d97e84842cfcf2406224eefdcea5dde5/charts/prometheus-node-exporter/templates/verticalpodautoscaler.yaml)

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-workflows/templates/server/server-deployment.yaml [hunk 1]
```diff
@@ -12,7 +12,9 @@ metadata:
     {{- toYaml . | nindent 4 }}
   {{- end }}
 spec:
+  {{- if (ne .Values.server.autoscaling.enabled true) }}
   replicas: {{ .Values.server.replicas }}
+  {{- end }}
```
Comment (yu-croco) [inline charts/argo-workflows/templates/server/server-deployment.yaml:17]: `replicas` is not supposed to be defined when autoscaling is enabled, maybe condition flag is opposite? 🤔 
```diff
-{{- if (ne .Values.server.autoscaling.enabled true) }}
+{{- if not .Values.server.autoscaling.enabled }}
  replicas: {{ .Values.server.replicas }}
{{- end }}
```

Comment (yu-croco) [reply charts/argo-workflows/templates/server/server-deployment.yaml:17]: exiting logic is https://github.com/argoproj/argo-helm/blob/main/charts/argo-cd/templates/argocd-server/deployment.yaml#L19-L21 .

Comment (agilgur5) [reply charts/argo-workflows/templates/server/server-deployment.yaml:17]: Ah good catch! The logic still works ("not equal true"), but yea the newer version is cleaner/simpler than the older. I'll update that

Comment (agilgur5) [reply charts/argo-workflows/templates/server/server-deployment.yaml]: updated / fixed

**Review by yu-croco — APPROVED** [review]

**Review by jmeridth — APPROVED** [review]

**Pull request merged**